ardupilot/equipment/power/20008.SmartBatteryContinuous.uavcan

```diff
@@ -0,0 +1,28 @@
+#
+# Smart battery data to be sent continuously (1 - 10Hz)
+#
+
+float16 temperature             # [C]      : Pack mounted thermistor (preferably installed between cells), NAN: field not provided
+float32 current                 # [Ampere] : Postive: defined as a discharge current. Negitive: deifined as a charging current, NAN: field not provided
+float16 voltage                 # [Volt]
+uint32 capacity_consumed        # [mAh]
+uint32 capacity_remaining       # [mAh]
```

Comment:
Why the change from WH?

Reply:
I agree W-h is a better standard from a power systems perspective. But in the end most flight controllers currently use mAh all around the code meaning you would still need mAh or convert from W-h using nominal voltage.

Reply:
yup I tend to favor units that are more practical, mAh is just what everyone uses

Reply:
👍

Reply:
@dakejahl : Shouldn't this change to WH moving forward though? Many drone makers are going towards higher(12S,14S or higher) packs, and the mAh drawn isn't as intuitive/accurate as when dealing with lower number cells. I feel like the better standards should also nudge users to adopting better units.  
You could check out https://rotoye.com/modelling-the-behavior-of-a-battery/ to see how the currents change with high and low SOC for different cell counts

Reply:
>However under load the voltage will sag. Is it OK to use the voltage while under load for the conversion? I'm not sure.

I believe that is a feasible option. It ignores the battery's internal resistance losses which may be <5% for high currents

Reply:
> ```
> Wh = mAh * Voltage / 1000
> ```
Using the design voltage gives a much more stable Wh figure. Otherwise when you let off on the throttle you will have a Wh increase. 

> > However under load the voltage will sag. Is it OK to use the voltage while under load for the conversion? I'm not sure.
> 
> I believe that is a feasible option. It ignores the battery's internal resistance losses which may be <5% for high currents

The figure that is given isn't just the pack's mAh's. Super simply it is the capacity that the BMS gauges went out of the pack over its previous cycle. I believe this is already ignoring the batteries internal losses.

Reply:
>Otherwise when you let off on the throttle you will have a Wh increase.

Yeah I was going to mention this, thought of it after I hit enter. I think it's a bad idea to try and convert units. Especially since everyone will just convert back to mAh to display/use the value.

Reply:
The way I've usually done the conversion is using nominal voltage? I'm not sure how else you could do it consistently for instantaneous measurements?

Reply:
> Wh = mAh * Voltage / 1000

This is not exactly right. To be precise, this should be `delta Wh = delta mAh * instantaneous voltage /1000`

`Present Wh left in battery = SOC(%) * Total Wh of battery`. 
But `Present mAh left in battery != SOC(%) * Total mAh of battery`

> > Otherwise when you let off on the throttle you will have a Wh increase.
> 
> Yeah I was going to mention this, thought of it after I hit enter. I think it's a bad idea to try and convert units. Especially since everyone will just convert back to mAh to display/use the value.

I strongly disagree. mAh is a hacky convention that was made for constant voltage batteries like AA, lead acid and is a bad one to follow in this DSDL.

Reply:
I am not opposed as long as I understand how we perform the conversion without losing any accuracy. How is `delta mAh` derived? Is this just `mAh_full - mAh_current`?

Reply:
Could you clarify? I didn't understand which variables are available, and which one you'd want to convert to. 

Analog power monitors for eg, have instantaneous current measurements (`current`). I believe px4 does `delta mAh` = `current`*`dt`

Reply:
> > Wh = mAh * Voltage / 1000
> 
> This is not exactly right. To be precise, this should be `delta Wh = delta mAh * instantaneous voltage /1000`
> 
> `Present Wh left in battery = SOC(%) * Total Wh of battery`. But `Present mAh left in battery != SOC(%) * Total mAh of battery`
> 
> > >

Most fuel gauges that I've looked at only report RemainingCapacity in units of mAh. This PR was intended specifically for smart batteries (which use fuel gauges). So if I can understand how we would convert this mAh value into Wh without losing accuracy then I'm game, otherwise the unit conversion would defeat the purpose of the data...

@alexklimaj Can you comment here?

Reply:
Even if the Wh isn't calculated by the chip, it maybe close enough to take volt at A and B and average it to calculate Wh from `delta mAh` .
![Wh calculation](https://user-images.githubusercontent.com/518314/163451757-d5df3688-e1b6-4ab4-a8bd-054331b130c2.jpg)

Reply:
I'm not sure how the `delta mAh` is helpful. Since this is smart battery specific, the fuel gauge IC's are doing coulomb counting based on current. Which I assume is why they measure everything in mAh as truth. I think generally, Watt Hours are calculated based on nominal voltage * Ah. If we want to use Wh, we would need to know the nominal voltage of the pack as well. 

We already have the nominal voltage in the Periodic message. In that case, mAh makes more sense. A user's application can take those two values and multiply them for Wh if they want.

In PX4, I believe `delta mAh` is just a rough running accumulation of the measured current. I've never actually used it as a real measure of the battery's remaining capacity.